core-bundle/src/Repository/CronJobRepository.php

```diff
@@ -0,0 +1,50 @@
+<?php
+
+declare(strict_types=1);
+
+/*
+ * This file is part of Contao.
+ *
+ * (c) Leo Feyer
+ *
+ * @license LGPL-3.0-or-later
+ */
+
+namespace Contao\CoreBundle\Repository;
+
+use Contao\CoreBundle\Entity\CronJob;
+use Doctrine\Bundle\DoctrineBundle\Repository\ServiceEntityRepository;
+use Doctrine\Common\Persistence\ManagerRegistry;
+use Doctrine\DBAL\Connection;
+
+/**
+ * @internal
+ *
+ * @method object|null findOneByName(string $name)
+ */
+class CronJobRepository extends ServiceEntityRepository
+{
+    /**
+     * @var Connection
+     */
+    private $connection;
+
+    public function __construct(ManagerRegistry $registry)
+    {
+        parent::__construct($registry, CronJob::class);
+
+        $this->connection = $registry->getConnection();
+    }
+
+    public function lockTable(): void
+    {
+        $table = $this->getClassMetadata()->getTableName();
+
+        $this->connection->exec("LOCK TABLES $table WRITE, $table AS t0 WRITE, $table AS t0_ WRITE");
```

Comment:
In the `RememberMeRepository`, this statement is just `"LOCK TABLES $table WRITE, $table AS t0_ WRITE"`. Why do we need the additional `t0` here? And why not just `"LOCK TABLES $table WRITE"`?

Reply:
That has already be answered, hasn't it? 😂 Doctrine automatically creates aliases for **every** table internally

Reply:
Yes, but why do we need `t0` here and not in the `RememberMeRepository` class?

Reply:
Otherwise I get this exception:

```
  [Doctrine\DBAL\Exception\DriverException]                                                                                                                                                           
  An exception occurred while executing 'SELECT t0.id AS id_1, t0.name AS name_2, t0.lastRun AS lastRun_3 FROM tl_cron_job t0 WHERE t0.name = ? LIMIT 1' with params ["Contao\\CoreBundle\\Cron\\Le   
  gacyCron::onMinutely"]:                                                                                                                                                                             
                                                                                                                                                                                                      
  SQLSTATE[HY000]: General error: 1100 Table 't0' was not locked with LOCK TABLES                                                                                                                     
                                                                                                                                                                                                      

Exception trace:
  at \vendor\doctrine\dbal\lib\Doctrine\DBAL\Driver\AbstractMySQLDriver.php:106
 Doctrine\DBAL\Driver\AbstractMySQLDriver->convertException() at \vendor\doctrine\dbal\lib\Doctrine\DBAL\DBALException.php:166
 Doctrine\DBAL\DBALException::wrapException() at \vendor\doctrine\dbal\lib\Doctrine\DBAL\DBALException.php:146
 Doctrine\DBAL\DBALException::driverExceptionDuringQuery() at \vendor\doctrine\dbal\lib\Doctrine\DBAL\Connection.php:912
 Doctrine\DBAL\Connection->executeQuery() at \vendor\doctrine\orm\lib\Doctrine\ORM\Persisters\Entity\BasicEntityPersister.php:718
 Doctrine\ORM\Persisters\Entity\BasicEntityPersister->load() at \vendor\doctrine\orm\lib\Doctrine\ORM\EntityRepository.php:193
 Doctrine\ORM\EntityRepository->findOneBy() at \vendor\doctrine\orm\lib\Doctrine\ORM\EntityRepository.php:311
 Doctrine\ORM\EntityRepository->resolveMagicCall() at \vendor\doctrine\orm\lib\Doctrine\ORM\EntityRepository.php:228
 Doctrine\ORM\EntityRepository->__call() at \vendor\contao\contao\core-bundle\src\Cron\Cron.php:85
 Contao\CoreBundle\Cron\Cron->run() at \vendor\contao\contao\core-bundle\src\Command\CronCommand.php:50
 Contao\CoreBundle\Command\CronCommand->execute() at \vendor\symfony\console\Command\Command.php:255
 Symfony\Component\Console\Command\Command->run() at \vendor\symfony\console\Application.php:1029
 Symfony\Component\Console\Application->doRunCommand() at \vendor\symfony\framework-bundle\Console\Application.php:97
 Symfony\Bundle\FrameworkBundle\Console\Application->doRunCommand() at \vendor\symfony\console\Application.php:272
 Symfony\Component\Console\Application->doRun() at \vendor\symfony\framework-bundle\Console\Application.php:83
 Symfony\Bundle\FrameworkBundle\Console\Application->doRun() at \vendor\symfony\console\Application.php:148
 Symfony\Component\Console\Application->run() at \vendor\contao\contao\manager-bundle\bin\contao-console:38
```

Reply:
Strange that it works in the `RememberMeRepository` without `t0`. But whatever. 😄

Reply:
🤷‍♂ 

It seems hacky to me to use `t0` just like that anyway. I mean we cannot really be certain about what Doctrine will actually use, can we?

Reply:
No, we cannot. 🤷‍♂ Does Doctrine not provide a method to lock tables?

Reply:
Why do you need locking on a table level? Is this only used to guarantee non-concurrent execution of the script?

Reply:
Yes, otherwise a cron job might be executed twice within the same minute for example.